tests/suites/helpers.function

```diff
@@ -193,6 +193,9 @@ static struct
     const char *test;
     const char *filename;
     int line_no;
+#if defined(MBEDTLS_TEST_MUTEX_USAGE)
+    const char *mutex_usage_error;
+#endif
```

Comment:
I did it this way so that the functional result is not suppressed. If a test case is failing because it's functionally broken, there may be “auxiliary” errors in that it misuses mutexes. It's easier to understand what's going on if the functional error is reported. For example, if a data structure is in the wrong state at some point in the middle of the test, it's typically easier to figure this out because a library function returned a `BAD_INPUT` error code, than because a mutex turns out to be used after freeing.

I do wish we could report multiple errors from a test case, but that's a long-term wish, not something that can be retrofitted easily. I don't see a problem with having specific fields in `test_info` for library/platform usage errors (now mutex, tomorrow memory, etc.).

Reply:
Thanks for the example. I think something like a `test_warn` would be of use here and although I'm still not particularly happy about this workaround at least it works for now.